zetaclient/bitcoin_client.go

```diff
@@ -815,12 +815,13 @@ func (ob *BitcoinChainClient) findNonceMarkUTXO(nonce uint64, txid string) (int,
 //
 // Parameters:
 //   - amount: The desired minimum total value of the selected UTXOs.
-//   - utxoCap: The maximum number of UTXOs to be selected.
+//   - utxos2Spend: The maximum number of UTXOs to spend.
 //   - nonce: The nonce of the outbound transaction.
+//   - csldtRank: The rank below which UTXOs will be consolidated.
 //   - test: true for unit test only.
 //
 // Returns: a sublist (includes previous nonce-mark) of UTXOs or an error if the qulifying sublist cannot be found.
-func (ob *BitcoinChainClient) SelectUTXOs(amount float64, utxoCap uint8, nonce uint64, test bool) ([]btcjson.ListUnspentResult, float64, error) {
+func (ob *BitcoinChainClient) SelectUTXOs(amount float64, utxos2Spend uint16, nonce uint64, csldtRank uint16, test bool) ([]btcjson.ListUnspentResult, float64, uint16, float64, error) {
```

Comment:
```suggestion
func (ob *BitcoinChainClient) SelectUTXOs(amount float64, utxos2Spend uint16, nonce uint64, consolidationRank uint16, test bool) ([]btcjson.ListUnspentResult, float64, uint16, float64, error) {
```
I think it is best to use explicit name for variables

Reply:
I can use `consolidateRank` to avoid name conflicting with the global constant `consolidationRank `